app/src/main/java/com/reminder/hydration/EntryActivity.kt

```diff
@@ -0,0 +1,74 @@
+package com.reminder.hydration
+
+import android.annotation.SuppressLint
+import android.graphics.Color
+import android.os.Bundle
+import android.widget.*
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+
+class EntryActivity : AppCompatActivity() {
+
+    private val sharedPrefManager by lazy {
+        SharedPrefManager.getInstance(this)
+    }
+
+    @SuppressLint("WrongViewCast")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_entry)
+
+        val waterReminderNotifGroup = findViewById<RadioGroup>(R.id.waterReminderNotifGroup)
+        val notificationSoundGroup = findViewById<RadioGroup>(R.id.notificationSoundGroup)
+        val entryPageSubmitButton = findViewById<Button>(R.id.entryPageSubmitButton)
+        val entryPageMessage = findViewById<TextView>(R.id.entryPageMessage)
+
+        val viewModel: EntryViewModel by viewModels {
+            EntryViewModelFactory(sharedPrefManager)
+        }
+
+        val wakeUpTime = findViewById<EditText>(R.id.wakeUpTimeH)
+        val sleepTime = findViewById<EditText>(R.id.sleepTimeH)
+        val waterNum = findViewById<EditText>(R.id.drunkenWaterAmount)
+
+        viewModel.loadData()
```

Comment:
load data should not be called here. View model should load data as soon as the objects associated with the data are initialized. Look at [this code snippet](https://developer.android.com/topic/libraries/architecture/viewmodel#implement) with a function similar to loadData...